src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureFlagKeyValueAdapter.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Azure.Data.AppConfiguration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.FeatureManagement
+{
+    internal class FeatureFlagKeyValueAdapter : IKeyValueAdapter
+    {
+        private static string _schemaVersion = FeatureManagementConstants.FeatureManagementDefaultSchema;
+
+        public FeatureFlagKeyValueAdapter(string schemaVersion)
+        {
+            if (schemaVersion != FeatureManagementConstants.FeatureManagementSchemaV1
+                && schemaVersion != FeatureManagementConstants.FeatureManagementSchemaV2)
+            {
+                throw new InvalidOperationException(
+                    $"Please select a valid Feature Management schema version by setting the '{FeatureManagementConstants.FeatureManagementSchemaEnvironmentVariable}' " + 
+                    $"environment variable to '{FeatureManagementConstants.FeatureManagementSchemaV1}' or '{FeatureManagementConstants.FeatureManagementSchemaV2}'.");
+            }
+
+            _schemaVersion = schemaVersion;
+        }
+
+        public Task<IEnumerable<KeyValuePair<string, string>>> ProcessKeyValue(ConfigurationSetting setting, CancellationToken cancellationToken)
+        {
+            FeatureFlag featureFlag;
+
+            try
+            {
+                featureFlag = JsonSerializer.Deserialize<FeatureFlag>(setting.Value);
+            }
+            catch (JsonException e)
+            {
+                throw new FormatException(setting.Key, e);
+            }
+
+            if (cancellationToken.IsCancellationRequested)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+            }
+
+            string featureFlagsSectionPrefix;
+
+            if (_schemaVersion == FeatureManagementConstants.FeatureManagementSchemaV1)
+            {
+                featureFlagsSectionPrefix = FeatureManagementConstants.FeatureManagementSectionName;
+            }
+            else
+            {
+                featureFlagsSectionPrefix = FeatureManagementConstants.FeatureManagementSectionName +
+                                            ConfigurationPath.KeyDelimiter +
```

Comment:
I see that we stuck with ':' everywhere else. Intentional to use `ConfigurationPath.KeyDelimiter` here?

Reply:
Also, this can be caculated in constructor and then only have to perform a single concatenation for the lifetime of the adapter.